service/history/api/recordactivitytaskstarted/api.go

```diff
@@ -42,6 +44,12 @@ func Invoke(
 	workflowConsistencyChecker api.WorkflowConsistencyChecker,
 	matchingClient matchingservice.MatchingServiceClient,
 ) (resp *historyservice.RecordActivityTaskStartedResponse, retError error) {
+	if activityRefProto := request.GetComponentRef(); activityRefProto != nil {
+		activityRef := chasm.ProtoRefToComponentRef(activityRefProto)
+
+		return activity.HandleRecordActivityTaskStarted(
```

Comment:
Repeating myself here but you can just change this to `chasm.UpdateComponent` with a method call on the `Activity` struct.